### ISSUE TITLE
"05:04:39 Jun 23, 2022 PDT" doesn't parse the timezone

### ISSUE THREAD

**Issue by cpearce** [dateutil/dateutil#1221]
If you parse the string `"05:04:39 Jun 23, 2022 PDT"` with `dateutil.parser.parse`, it doesn't come out with a timezone on the return value. I'd expect that to be in timezone UTC-7.

Testcase:

```python
from dateutil.parser import parse

d = parse("05:04:39 Jun 23, 2022 PDT")
print(f'parse("05:04:39 Jun 23, 2022 PDT") = {d}')
print(f"d.tzinfo = {d.tzinfo} (shouldn't this be UTC-7?)")
```
Outputs...
```
parse("05:04:39 Jun 23, 2022 PDT") = 2022-06-23 05:04:39
d.tzinfo = None (shouldn't this be UTC-7?)
```

Tested with python-dateutil version 2.8.2, Python 3.9.7, Linux.

**Comment by jbrockmendel** [comment]
"PDT" isn't, strictly speaking, a timezone.  If you pass `tzinfos={"PDT": dateutil.tz.gettz("US/Pacific")}` you should get what you want

**Comment by cpearce** [comment]
I'm not disagreeing with you; we're getting this string passed into our system from users, who put the most random stuff in their datetime strings.

Thanks for pointing out the work around; that's helpful, and it lets us get past this!

**Comment by dlm6693** [comment]
Circling back on this. I'm seeing that a warning is being raised now for `PDT`.
```python
from dateutil import parser

parser.parse('Wed May 20 13:51:10 PDT 2015')

UnknownTimezoneWarning: tzname CDT identified but not understood.  Pass `tzinfos` argument in order to correctly return a timezone-aware datetime.  In a future version, this will raise an exception.
warnings.warn("tzname {tzname} identified but not understood.  "

datetime.datetime(2015, 5, 20, 13, 51, 10)
```

I understand that `PDT` is not really a timezone, but nor is `EDT` and I see see no such warning. Is this expected behavior? Should we be mapping all pseudo daylight savings timezones to their "real" counterparts?

**Comment by jbrockmendel** [comment]
> I understand that PDT is not really a timezone, but nor is EDT and I see see no such warning. Is this expected behavior?

Assuming you are running this on a machine whose system timezone is US/Eastern, yes (specifically, if `"EDT" in time.tzname`).  This is a behavior that I would be +1 on deprecating (and we have deprecated it in pandas)

**Comment by dlm6693** [comment]
>Assuming you are running this on a machine whose system timezone is US/Eastern, yes (specifically, if "EDT" in time.tzname). This is a behavior that I would be +1 on deprecating (and we have deprecated it in pandas)

Yep exactly should have mentioned I'm running this locally and am based in the eastern timezone. It appears in this situation the tzinfo gets set to `tzlocal()`, which makes sense, but is kind of a halfway solution. Daylight savings timezones should either be mapped completely or not at all. I'm guessing the latter is a safer choice for this library.

**Comment by jaraco** [comment]
> "PDT" isn't, strictly speaking, a timezone. If you pass `tzinfos={"PDT": dateutil.tz.gettz("US/Pacific")}` you should get what you want

Unfortunately, that doesn't get me what I want, which is an approach that generalizes to date times that are found in the wild. For example, I see these date times when working with twilio data:

![image](https://github.com/user-attachments/assets/342b7f73-0f9d-4495-a257-1816b1b8ae8e)

And I imagine there are other applications out there that generate dates with time zones other than PDT and PST. Has anyone out there generated a library that implements what `tzinfos` needs to parse these common formats?

**Comment by jaraco** [comment]
Gemini [helped me put together such a list](https://g.co/gemini/share/6b38d8b22c2f) and I've published it in [tempora 5.7.0](https://pypi.org/project/tempora).

Now one can use `tempora.tzinfos` for common tzinfos or use `tempora.parse` to invoke `dateutil.parser.parse` with the tzinfos.
